src/otx/core/model/entity/visual_prompting.py

```diff
@@ -225,6 +256,90 @@ def _customize_outputs(
             labels=[torch.cat(list(labels.values())) for labels in inputs.labels],
         )
 
+    def optimize(  # type: ignore[override]
+        self,
+        output_dir: Path,
+        data_module: OTXDataModule,
+        ptq_config: dict[str, Any] | None = None,
+    ) -> dict[str, Path]:
+        """Runs NNCF quantization."""
+        import nncf
+        import openvino
+
+        def check_if_quantized(model: openvino.Model) -> bool:
+            """Checks if OpenVINO model is already quantized."""
+            nodes = model.get_ops()
+            return any(op.get_type_name() == "FakeQuantize" for op in nodes)
+
+        def transform_fn(
+            data_batch: VisualPromptingBatchDataEntity | ZeroShotVisualPromptingBatchDataEntity,
+            module: Literal["image_encoder", "decoder"],
+        ) -> np.ndarray | dict[str, Any]:
+            images, _, prompts = self._customize_inputs(data_batch)  # type: ignore[arg-type]
+
+            image = images[0]["images"]  # use only the first image
+            if module == "image_encoder":
+                # resize
+                resized_image = self.model["image_encoder"].resize(
+                    image[0],
+                    (self.model["image_encoder"].w, self.model["image_encoder"].h),
+                )
+
+                # pad image if necessary because `fit_to_window` resize for python in modelapi doesn't support pad
```

Comment:
ModelAPI support resize + pad by 'fit_to_window_letterbox'. Honestly, I don't know what it is for exactly. But should this part be changed if ModelAPI can support pad?

Reply:
I reported the issues. https://github.com/openvinotoolkit/model_api/issues/167
`fit_to_window` with numpy doesn't do pad after resize, but one with opset does pad after resize.
So, if we want to use numpy version, padding after reisze is required.